spec/controllers/automated_tests_controller_spec.rb

```diff
@@ -548,8 +549,79 @@
     let(:role) { create(:student) }
 
     context 'GET student_interface' do
-      before { get_as role, :student_interface, params: params }
-      # TODO: write tests
+      let(:assignment) do
+        create(:assignment,
+               assignment_properties_attributes: {
+                 enable_student_tests: true,
+                 tokens_per_period: 5,
+                 token_start_date: 1.day.ago,
+                 token_period: 24
+               })
+      end
+      let(:grouping) { create(:grouping, assignment: assignment) }
+      let(:student) { create(:student) }
+      let(:role) { student }
+      let(:params) { { course_id: assignment.course.id, assignment_id: assignment.id } }
+
+      before do
+        allow_any_instance_of(AutotestSetting).to(
+          receive(:send_request!).and_return(OpenStruct.new(body: { api_key: 'someapikey' }.to_json))
+        )
+        create(:student_membership, role: student, grouping: grouping, membership_status: 'accepted')
+        sign_in student
+      end
+
+      context 'when student tests are enabled and the student has a grouping' do
+        before do
+          get_as student, :student_interface, params: params
+        end
+
+        it 'should calculate the next token generation time' do
+          expect(assigns(:next_token_generation_time)).not_to be_nil
```

Comment:
This test should be more specific. Calculate the exact time that should be displayed. This is related to my above comment about this time should actually be generated.